test/test_router.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"net/http"
+	"time"
+
+	c "github.com/flynn/go-check"
+)
+
+type RouterSuite struct {
+	Helper
+}
+
+var _ = c.ConcurrentSuite(&RouterSuite{})
+
+func (s *RouterSuite) TestAdditionalHttpPorts(t *c.C) {
+	// boot 1 node cluster
+	x := s.bootCluster(t, 1)
+	defer x.Destroy()
+
+	// update router env
+	t.Assert(x.flynn("/", "-a", "router", "env", "set", "ADDITIONAL_HTTP_PORTS=8080"), Succeeds)
+	time.Sleep(1 * time.Second)
+
+	// check a non-routed HTTP request to an additional port fails
+	req, err := http.NewRequest("GET", "http://dashboard."+x.Domain+":8080", nil)
+	t.Assert(err, c.IsNil)
+	req.SetBasicAuth("", x.Key)
+	res, err := http.DefaultClient.Do(req)
+	t.Assert(err, c.IsNil)
+	t.Assert(res.StatusCode, c.Equals, http.StatusNotFound)
+
+	// add a controller route on the new port
+	t.Assert(x.flynn("/", "-a", "dashboard", "route", "add", "http", "-s", "dashboard-web", "-p", "8080", "dashboard."+x.Domain), Succeeds)
+	time.Sleep(1 * time.Second)
```

Comment:
This appears to be due to an issue with `httpSyncHandler.Set` not being called synchronously within `AddRoute`, resulting in a state where the route is in the database, but has not populated `HTTPListener.domains` in time for the `findRoute` call. This issue likely existed prior to my changes, but was exposed by the structure of my test.

A delay avoids this existing issue in the new test, allowing this code (which does not affect this) to move forward. I can make a separate ticket for the other issue, if you'd like.